oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java

```diff
@@ -81,4 +83,69 @@ public static <T> Set<T> toSet(final Iterator<T> iterator) {
         return result;
     }
 
-}
+    /**
+     * Convert a vararg list of items to a set.  The returning set is mutable and supports all optional operations.
+     * @param elements elements to convert
+     * @return the set
+     * @param <T> the type of the elements
+     */
+    @SafeVarargs
+    public static <T> Set<T> toSet(final T... elements) {
+        final Set<T> result = new HashSet<>(elements.length);
```

Comment:
Here you are specifying the `capacity` of the set, which in case of HashSet and HashMap works in a different way than arrays. The HashSet will ensure that `numberOfElements < capacity*loadFactor`. So setting the capacity to the number of elements that will be inserted here, guarantees that there will be a resizing operation. So probably use a capacity of `elements.length * 1.25` (the default load factor is 0.75)

Reply:
or specify the load factor as 1?

Reply:
I'm not sure about the performance implications of using load factor of 1, I guess it will increase collisions which may degrade performance, of both inserts and lookups. As the returned map is mutable, we can expect that it will modified, so I think it's safer to keep the default load factor, which I guess was carefully chosen by very clever people to provide good performance. If we create the map with load factor of 1 and then it is used for a performance critical operation, we risk that the map will perform more poorly than if it had been created with the default load factor.